src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -33,162 +43,465 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessageDelegateParameterMoreMeaningfulName,
+            DiagnosticCategory.Naming,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false,
+            description: s_localizableDescription,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1704",
+            customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        internal static DiagnosticDescriptor TypeTypeParameterMoreMeaningfulNameRule = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessageTypeTypeParameterMoreMeaningfulName,
+            DiagnosticCategory.Naming,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false,
+            description: s_localizableDescription,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1704",
+            customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        internal static DiagnosticDescriptor MethodTypeParameterMoreMeaningfulNameRule = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessageMethodTypeParameterMoreMeaningfulName,
+            DiagnosticCategory.Naming,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false,
+            description: s_localizableDescription,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1704",
+            customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            FileParseRule,
+            AssemblyRule,
+            NamespaceRule,
+            TypeRule,
+            VariableRule,
+            MemberRule,
+            MemberParameterRule,
+            DelegateParameterRule,
+            TypeTypeParameterRule,
+            MethodTypeParameterRule,
+            AssemblyMoreMeaningfulNameRule,
+            NamespaceMoreMeaningfulNameRule,
+            TypeMoreMeaningfulNameRule,
+            MemberMoreMeaningfulNameRule,
+            MemberParameterMoreMeaningfulNameRule,
+            DelegateParameterMoreMeaningfulNameRule,
+            TypeTypeParameterMoreMeaningfulNameRule,
+            MethodTypeParameterMoreMeaningfulNameRule);
+
         public override void Initialize(AnalysisContext analysisContext)
-#pragma warning restore RS1025 // Configure generated code analysis
         {
             analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext compilationStartContext)
```

Comment:
As far as I can tell from GitHub, this could be marked as `static`.

Reply:
Done